### PR TITLE
Updated custom builder description

### PR DESCRIPTION
**Pull request by soltysh**
@bparees as promised in [here](https://github.com/openshift/openshift-docs/pull/492#discussion_r32443358) I'm updating the description. ptal

@adellape fyi

### CODE REVIEW

**Comment by bparees** [comment]
lgtm for now, but ultimately needs more fleshing out about what is really expected of a custom builder (read the build definition env variable, do whatever you want to do, output artifacts to the build's output location if one is defined, etc)

**Comment by soltysh** [comment]
@bparees I've added a few words about how the process should look like, ptal.

**Comment by adellape** [comment]
Pending ack from @bparees, I'm good with merging this and following up with some minor edits to wording.

File: creating_images/custom.adoc [hunk 1]
```diff
@@ -38,3 +40,13 @@ the information needed to proceed with the build:
...
+
+== Custom builder workflow
+Although custom builder image author has a great flexibility in defining the build
+process on its own, still he should follow a few required steps necessary to seamlessly
+run a build inside of OpenShift. The required steps for a custom builder image are following:
+
+. Read the link:../rest_api/openshift_v1.html#v1-build[Build] definition, which
+contains all the necessary information about input parameters for the build.
+. Run the build process.
+. Output the artifacts to the link:../rest_api/openshift_v1.html#v1-build[Build]'s output location if one is defined.
```
Comment (bparees) [inline creating_images/custom.adoc]: this really needs to say "push the image to the output location if the build produces an image and if the output location is defined"

our output is not (currently) a generic destination.     Whether in the future we want to make it one, or we just say "if you want to output to an ftp server, define an env variable in your buildconfig that the custom builder can look at to know where to upload to" i don't particularly care.

Comment (soltysh) [inline creating_images/custom.adoc]: Updated and created issue https://github.com/openshift/origin/issues/3231

File: creating_images/custom.adoc [hunk 2]
```diff
@@ -38,3 +40,14 @@ the information needed to proceed with the build:
...
+
+== Custom builder workflow
+Although custom builder image author has a great flexibility in defining the build
+process on its own, still he should follow a few required steps necessary to seamlessly
+run a build inside of OpenShift. The required steps for a custom builder image are following:
+
+. Read the link:../rest_api/openshift_v1.html#v1-build[Build] definition, which
+contains all the necessary information about input parameters for the build.
+. Run the build process.
+. Push the image to the link:../rest_api/openshift_v1.html#v1-build[Build]'s output location if the
```
Comment (bparees) [inline creating_images/custom.adoc]: you're still assuming the custom builder builds an image.  

this should be "if the product of the build is a new image, push the image to...."

Comment (soltysh) [inline creating_images/custom.adoc]: Is that better now?

**Comment by bparees** [comment]
lgtm.

**Comment by soltysh** [comment]
@adellape merge?

**Pull request merged**